app/cdap/components/PipelineSummary/RunsGraphHelpers.js

```diff
@@ -174,12 +174,15 @@ export function getDuration(time) {
 }
 
 export function getGapFilledAccumulatedData(data, numOfDataPoints) {
-  let { x: minx, y: miny } = data[0];
-  let maxx = data[data.length - 1].x;
-  let numberOfEntries = maxx - minx;
+  const { x: minx, y: miny } = data[0];
+  const maxx = data[data.length - 1].x;
+  const numberOfEntries = maxx - minx;
   let lasty = miny;
-  let finalData = Array(numberOfEntries + 1).map((i, index) => {
-    let matchInActualData = data.find((d) => d.x === minx + index);
+  // Create an array with numberOfEntries + 1 undefined values
+  const arrayOfSize = [...Array(numberOfEntries + 1)];
```

Comment:
This is the fix for the reported issue. Spreading the empty array creates an array of N `undefined` entries.